### DISCUSSION TITLE
Stats invalid when performing aggregate query and normal query lazily

### DISCUSSION THREAD

**Discussion by nickchampion** [ravendb/ravendb#14146] [General]
HI, 

I am trying to do a faceted search, I am building up a query, calling lazily, applying the aggregations and then execute the aggregations query lazily as follows, this is using the Node.js client.

Code as follows

```
const apply_aggregations = (search_context: SearchContext): Lazy<FacetResultObject> => {
  search_context.facet_query = search_context.query.aggregateBy(search_context.facets[0])

  search_context.facets.slice(1).forEach(f => {
    search_context.facet_query = search_context.facet_query.andAggregateBy(f)
  })

  return search_context.facet_query.executeLazy()
}

let stats: QueryStatistics = null

const lazy_results = search_context.query.statistics(s => (stats = s)).lazily()
const lazy_aggregation_results = apply_aggregations(search_context)

const results = await lazy_results.getValue()
search_context.facet_results = await lazy_aggregation_results.getValue()
```

Now there are 3 documents returned by this query, I am applying 7 facets, the totalResults field on statistics however is showing 7 so it looks like its pulling back the stats for the facet query rather than the initial document query.

If I don't use the Lazily API I can get the correct result count before issuing the second query, but when I do this lazily the stats always reflects the aggregation query stats not the document query stats.

Is there some way to extract the stats from the first lazy query?

Thanks

**Comment by garayx** [comment]
Hi 

Can you send a unit-test or a sample project that reproduce the issue?
[Example for unit-test](https://github.com/ravendb/ravendb-nodejs-client/blob/b8e6cf6e822259a545f6103608a8f67113c78478/test/Issues/RavenDB_11770.ts#L10)

**Reply by nickchampion** [reply]
Hi, I have opened an issue

https://github.com/ravendb/ravendb/issues/14149

and a PR with a failing test

https://github.com/ravendb/ravendb-nodejs-client/pull/307

Thanks
